src/Cache/ApcuStore.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace TusPhp\Cache;
+
+use Carbon\Carbon;
+use APCUIterator;
+
+class ApcuStore extends AbstractCache
+{
+
+    /**
+     * {@inheritDoc}
+     */
+    public function get(string $key, bool $withExpired = false)
+    {
+        $contents = apcu_fetch($this->getActualCacheKey($key)) ?: null;
+
+        if ($withExpired) {
+            return $contents ?: null;
+        }
+
+        $isExpired = Carbon::parse($contents['expires_at'])->lt(Carbon::now());
+
+        return $isExpired ? null : $contents;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function set(string $key, $value)
+    {
+        $contents = $this->get($key) ?: [];
+
+        if (\is_array($value)) {
+            $contents = $value + $contents;
+        } else {
+            array_push($contents, $value);
+        }
+
+        return apcu_store($this->getActualCacheKey($key), $contents, $this->getTtl());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function delete(string $key): bool
+    {
+        return apcu_delete($this->getActualCacheKey($key));
```

Comment:
Just to make sure that the return type is always boolean and make scrutinizer happy (probably), we can do:

```suggestion
        return true === apcu_delete($this->getActualCacheKey($key));
```

Reply:
Should not be a problem since `$key` is a string, but I too love a happy scrutinizer. 😄